### ISSUE TITLE
Shellcheck does not recognize !#

### ISSUE THREAD

**Issue by ZxMYS** [koalaman/shellcheck#1215]
#### For bugs
- Rule Id: SC1072
- My shellcheck version: online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/bash
exec scala "$0" "$@"
!#
println("test " + args.toList)

```

#### Here's what shellcheck currently says:

Line 3:
!#
  ^-- SC1072: Expected whitespace. Fix any mentioned problems and try again.

#### Here's what I wanted or expected to see:

!# indicates close of the header, and it should not trigger SC1072
https://stackoverflow.com/questions/10060419/what-is-the-meaning-of-bang-pound-in-a-sh-bash-shell-script?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

Even if you add `# shellcheck disable=SC1072` it still gives you that warning.

Plus, style check should be disabled for the lines after exec, because it's going to be Scala code. (can we have a global disable flag that disable all rules in a file?)

**Comment by koalaman** [comment]
#1201 requests the same kind of feature. `!#` in general is a Scala thing, and not to do with any shell that ShellCheck supports.
